qa/tasks/cephfs/test_exports.py

```diff
@@ -175,3 +175,131 @@ def test_session_race(self):
         # Check if rank1 changed (standby tookover?)
         new_rank1 = self.fs.get_rank(rank=1)
         self.assertEqual(rank1['gid'], new_rank1['gid'])
+
+class TestKillPoints(CephFSTestCase):
+    # Two active MDS
+    MDSS_REQUIRED = 2
+
+    init = False
+
+    def _setup_cluster(self):
+        # Set Multi-MDS cluster
+        self.fs.set_max_mds(2)
+
+        self.fs.wait_for_daemons()
+        
+        all_daemons = self.fs.get_daemon_names()
+
+        # Create test data
+        if self.init is False:
+            self._populate_data(8)
+            self.init = True
+
+        return True
+
+    def _populate_data(self, nfiles):
+            size_mb = 8
+            dname = "abc"
+            self.mount_a.run_shell(["mkdir", dname])
+            for i in range(nfiles):
+                fname =dname.join("/").join(str(i)).join(".txt")
+                self.mount_a.write_n_mb(fname, size_mb)
+
+            # Get the list of file to confirm the count of files
+            self.org_files = self.mount_a.ls()
+
+    def _verify_data(self):
+        s1 = set(self.mount_a.ls())
+        s2 = set(self.org_files)
+
+        if s1.isdisjoint(s2):
+            log.error("Directory contents org: %s" %str(org_files))
+            log.error("Directory contents new: %s" %str(out))
+            return False
+
+        log.info("Directory contents matches")
+        return True
+
+    def _run_export_dir(self, importv, exportv):
+        # Wait till all MDS becomes active
+        self.fs.wait_for_daemons()
+
+        # Get all active ranks
+        ranks = self.fs.get_all_mds_rank()
+
+        rank_0 = self.fs.get_rank(rank=0)
+        rank_0_name = rank_0['name']
+
+        killpoint = {}
+        killpoint[0] = ('export', exportv)
+        killpoint[1] = ('import', importv)
+
+        command = ["config", "set", "mds_kill_export_at", str(exportv)]
+        result = self.fs.mds_asok(command, rank_0_name)
+        assert(result["success"])
+
+        command = ["config", "set", "mds_kill_import_at", str(importv)]
+        result = self.fs.mds_asok(command, rank_1_name)
+        assert(result["success"])
+
+        # This should kill either or both MDS process
+        command = ["export", "dir", "/abc", "1"]
```

Comment:
Better to use an export pin so that when the failed rank comes back up, the export is retried. This will then trip the second killpoint (either import or export). BUT, eventually, the export should complete!